yaml-schema.json

```
@@ -0,0 +1,137 @@
+                                "type": "string",
+                                "description": "Type of the input channel"
+                            },
+                            "description": {
+                                "type": "string",
+                                "description": "Description of the input channel"
+                            },
+                            "pattern": {
+                                "type": "string",
+                                "description": "Pattern of the input channel"
+                            }
+                        },
+                        "required": ["type", "description"]
+                    }
+                }
+            }
+        },
+        "tools": {
+            "type": "array",
+            "description": "Tools used by the module",
+            "items": {
+                "type": "object",
+                "patternProperties": {
+                    ".*": {
+                        "type": "object",
+                        "properties": {
+                            "description": {
+                                "type": "string",
+                                "description": "Description of the output channel"
+                            },
+                            "homepage": {
+                                "type": "string",
+                                "description": "Homepage of the tool"
+                            },
+                            "documentation": {
+                                "type": "string",
+                                "description": "Documentation of the tool"
+                            },
+                            "tool_dev_url": {
+                                "type": "string",
+                                "description": "URL of the development version of the tool's documentation"
+                            },
+                            "doi": {
+                                "type": "string",
+                                "description": "DOI of the tool"
+                            },
+                            "licence": {
+                                "type": ["array", "string"],
+                                "description": "Licence of the tool",
+                                "items": {
+                                    "type": "string"
+                                }
+                            }
+                        },
+                        "required": ["description"],
+                        "anyOf": [
+                            { "required": ["homepage"] },
+                            { "required": ["documentation"] },
+                            { "required": ["tool_dev_url"] },
+                            { "required": ["doi"] }
+                        ]
+                    }
+                }
+            }
+        }
+    },
+    "required": ["name", "description", "keywords", "authors", "output", "tools"]
```

Isn't input required too?
```suggestion
    "required": ["name", "description", "keywords", "authors", "output", "input", "tools"]
```

---

Nope, there a some tools which don't have an input (usually the download their needed data), e.g. https://github.com/nf-core/modules/blob/master/modules/nf-core/krona/kronadb/meta.yml (I switched the input there to a comment)

---

But this is maybe also not the optimal solution, because one might be confused about that. Maybe I will show a standard message on the website if there is no input needed.

---

Oh, ok I see 👍